# 대륙별 음주데이터 분석

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
drinks = pd.read_csv("./data/drinks.csv")

# 데이터 파악하기!

In [ ]:
# 처음 5개 행 (이런~~ 데이터가 있구나하고 그냥 확인용)
drinks.head()

,country,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol,continent
0,Afghanistan,0,0,0,0.0,AS
1,Albania,89,132,54,4.9,EU
2,Algeria,25,0,14,0.7,AF
3,Andorra,245,138,312,12.4,EU
4,Angola,217,57,45,5.9,AF


In [ ]:
# 데이터프레임 행 수 / 열 수
drinks.shape

(193, 6)

In [ ]:
# 데이터프레임 각 열의 타입 보기
drinks.dtypes

country                          object
beer_servings                     int64
spirit_servings                   int64
wine_servings                     int64
total_litres_of_pure_alcohol    float64
continent                        object
dtype: object

In [ ]:
# 데이터프레임 drinks 열 정보, 데이터 타입, 결측치 수 / 정보확인
drinks.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   country                       193 non-null    object 
 1   beer_servings                 193 non-null    int64  
 2   spirit_servings               193 non-null    int64  
 3   wine_servings                 193 non-null    int64  
 4   total_litres_of_pure_alcohol  193 non-null    float64
 5   continent                     170 non-null    object 
dtypes: float64(1), int64(3), object(2)
memory usage: 9.2+ KB


In [ ]:
# 데이터프레임 수치형 열을 나타낸것 -> [평균,표준편차,최솟값,4분위수,최댓값]
drinks.describe()

,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol
count,193.000000,193.000000,193.000000,193.000000
mean,106.160622,80.994819,49.450777,4.717098
std,101.143103,88.284312,79.697598,3.773298
min,0.000000,0.000000,0.000000,0.000000
25%,20.000000,4.000000,1.000000,1.300000
50%,76.000000,56.000000,8.000000,4.200000
75%,188.000000,128.000000,59.000000,7.200000
max,376.000000,438.000000,370.000000,14.400000


# 전처리 시작!

In [ ]:
# 컬럼별 결측치의 개수 : continet열은 결측치 23개
drinks.isna().sum()

country                          0
beer_servings                    0
spirit_servings                  0
wine_servings                    0
total_litres_of_pure_alcohol     0
continent                       23
dtype: int64

In [ ]:
# 1) drinks["continent"] 하면 DataFrame drinks에서 "continet" 열(series)를 선택
# 2) drinks["continent"].unique 는 unique 함수가 series에서 중복 값을 제거하고 각 값 1차원 numpy 배열로 돌려준다.

drinks["continent"].unique()

array(['AS', 'EU', 'AF', nan, 'SA', 'OC'], dtype=object)

In [ ]:
# continent 열의 결측치 행만 골라서 drinks 데이터프레임으로 반환
# 참고: isna() 는 데이터 결측치 여부 확인해준다 -> 결측치 여부를 불리언 값으로 반환 -> 결측치면 True, 아니면 Fasle
# 결론: drinks["continent"].isna() 를 통해 결측치 여부 확인 -> 한 번 더 drinks[""] 해주면 결측치가 True 값만 반환
# 불리언 시리즈 넣은 이유 for True 행만 골라서 DF 반환하려고
# 그냥 제가 공부한거 필기하려고 이렇게 구구절절 적은거니까 첫 번째 주석위주로 보시면 됩니다.
drinks[drinks["continent"].isna()]

,country,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol,continent
5,Antigua & Barbuda,102,128,45,4.9,NaN
11,Bahamas,122,176,51,6.3,NaN
14,Barbados,143,173,36,6.3,NaN
17,Belize,263,114,8,6.8,NaN
32,Canada,240,122,100,8.2,NaN
41,Costa Rica,149,87,11,4.4,NaN
43,Cuba,93,137,5,4.2,NaN
50,Dominica,52,286,26,6.6,NaN
51,Dominican Republic,193,147,9,6.2,NaN
54,El Salvador,52,69,2,2.2,NaN


In [ ]:
# 결측치에 NA를 널는다
drinks["continent"] = drinks["continent"].fillna("NA")

In [ ]:
# *참고용* 발표할 때 "결측치였던 5번행이 NA로 대체된 것을 볼 수 있다"
drinks["continent"].iloc[5]

'NA'

In [ ]:
drinks["continent"].unique()

array(['AS', 'EU', 'AF', 'NA', 'SA', 'OC'], dtype=object)

In [ ]:
# 피피티 또는 발표 추가 참고 내용

# 전처리 한 후에 다시 isna로 결측치 확인 -> 아무것도 변환되지 않으면 결측치 "NA"로 대체 성공
drinks[drinks["continent"].isna()]

,country,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol,continent


In [ ]:
drinks.head()

,country,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol,continent
0,Afghanistan,0,0,0,0.0,AS
1,Albania,89,132,54,4.9,EU
2,Algeria,25,0,14,0.7,AF
3,Andorra,245,138,312,12.4,EU
4,Angola,217,57,45,5.9,AF


In [ ]:
# continent 열에 들어 있는 고유값이 몇 번 나타나는지 세어서, 내림차순으로 정렬한 serious를 반환한 것이다. 그저 참고
drinks["continent"].value_counts()

continent
AF    53
EU    45
AS    44
NA    23
OC    16
SA    12
Name: count, dtype: int64

# 상관관계 분석 !

### 다중 상관 분석

In [ ]:
drinks.columns

Index(['country', 'beer_servings', 'spirit_servings', 'wine_servings',
       'total_litres_of_pure_alcohol', 'continent'],
      dtype='object')

- 컬럼 설명

- beer_servings: 1인당 연간 맥주 제공량

- spirit_servings: 1인당 연간 증류주 제공량

- wine_servings: 1인당 연간 와인 제공량

- total_litres_of_pure_alcohol: 1인당 연간 순수 알코올 섭취량

In [ ]:
# cor_cols = 상관관계 나타낼 "컬럼의" 변수
cor_cols = drinks.columns[1:5]
cor_cols

Index(['beer_servings', 'spirit_servings', 'wine_servings',
       'total_litres_of_pure_alcohol'],
      dtype='object')

In [ ]:
# 1 - 상관관계 - 산점도

# drinks_corr에 넣어서 맥주, 증류주, 와인, 국가별 1인당 연간 순수 알코올 소비량 의 상관관계 나타냄
drinks_corr = drinks[cor_cols].corr()
drinks_corr

,beer_servings,spirit_servings,wine_servings,total_litres_of_pure_alcohol
beer_servings,1.000000,0.458819,0.527172,0.835839
spirit_servings,0.458819,1.000000,0.194797,0.654968
wine_servings,0.527172,0.194797,1.000000,0.667598
total_litres_of_pure_alcohol,0.835839,0.654968,0.667598,1.000000


#### 발표 참고용 (상관관계 설명, 산점도 그래프에 포함될 내용임)

- 상관관계를 분석하여, 각 주류 소비량이 총 알코올 섭취량에 어떠한 영향을 주는지 파악!
- 분석 대상은 국가별 음주 데이터이며, 상관계수 계산을 통해 변수 간 연관성 정도를 정량적으로 확인하였다


1) beer_servings ↔ total_litres_of_pure_alcohol (0.8358)
맥주 제공량과 "국가별 1인당 연간 순수 알코올 소비량" 간 상관계수는 약 0.836(2개의 상관성이 가장 높음), 네 변수 중 가장 높은 수치를 보임. 즉, 맥주 소비가 많은 국가일수록 전체 알코올 섭취량도 높다는 강한 양의 상관관계를 나타냄.

2) wine_servings ↔ total_litres_of_pure_alcohol (0.6676)
와인 제공량과 순수 알코올 소비량 간 상관계수는 약 0.668, 중간 이상의 양의 상관관계를 보인다. 와인 소비가 늘어날수록 순수 알코올 소비량 역시 증가하는 경향 확인 가능.

3) spirit_servings ↔ total_litres_of_pure_alcohol (0.6550)
증류주 제공량과 순수 알코올 소비량 간 상관계수는 약 0.655, 와인과 유사한 수준의 중간 양의 상관관계를 가진다. 증류주 소비량이 많을수록 전체 알코올 섭취량이 높아지는 경향 확인 가능.

4) beer_servings ↔ wine_servings (0.5272)
맥주와 와인 제공량 간 상관계수는 약 0.527로, 보통 수준의 양의 상관관계를 시사한다. 즉, 맥주 소비가 많은 국가일수록 와인 소비도 상대적으로 많은 편.

5) beer_servings ↔ spirit_servings (0.4588)
맥주와 증류주 제공량 간 상관계수는 약 0.459로, 약한 ~ 중간 수준의 양의 상관관계를 보인다. 맥주 소비가 증가하면 증류주 소비도 일부 연동되는 경향이 있으나, 와인과의 관계보다는 약함

6) wine_servings ↔ spirit_servings (0.1948)
와인과 증류주 제공량 간 상관계수는 약 0.195로, 매우 낮은 수준. 이는 와인 소비가 많다고 해서 증류주 소비가 반드시 많지 않음을 의미함. 두 변수 간의 상관관계가 거의 없다는 것을 나타냄.


=> total_litres_of_pure_alcohol 피처가 대체적으로 다른 모든 피처와 강한 상관관계가 있는 것으로 보임
국가별 음주 지표 네 변수 간 피어슨 상관관계를 계산하여, 맥주 제공량이 국가별 1인당 연간 순수 알코올 소비량에 가장 큰 영향을 미치며(0.8358), 그다음으로 와인(0.6676), 증류주(0.6550) 순임을 확인했다.


# 데이터 탐색하기

## 각 대륙별 비율 확인
-> 파이차트 시각화와 같은 내용

In [ ]:
# 2 - 대륙별 국가 수 비율 - 파이

continents = drinks["continent"].value_counts()

In [ ]:
continents

continent
AF    53
EU    45
AS    44
NA    23
OC    16
SA    12
Name: count, dtype: int64

In [ ]:
# 총 서빙 합계 컬럼 생성 - beer_servings + spirit_servings + wine_servings
# total_servings 라는 새로운 열을 형성 한 데이터프레임임

drinks['total_servings'] = ( drinks['beer_servings'] + drinks['spirit_servings'] + drinks['wine_servings'])

In [ ]:
# 3 - 대륙별 총 서빙량 - 막대 + 평균값
continent_total_servings = (drinks.groupby('continent')['total_servings'].sum().reset_index())

In [ ]:
continent_total_servings

,continent,total_servings
0,AF,4986
1,AS,4706
2,EU,21085
3,NA,7721
4,OC,2940
5,SA,4227


In [ ]:
# 총 서빙량의 평균값
# continent_total_servings 데이터 프레임에서 total_servings(3개 알코올 더한)을 평균으로 구한 것
avg_servings = continent_total_servings['total_servings'].mean()

In [ ]:
avg_servings

7610.833333333333

In [ ]:
# 4 - 대륙별 주종별 서빙량 - 막대그래프

# 이건 위에 코드 적어둔거
cor_cols = drinks.columns[1:4]
cor_cols

# 3개의 컬럼을 기준으로 대륙별로 그룹화하여 합계를 구한 것임
continents_sum = drinks[cor_cols].groupby(drinks["continent"]).sum()

In [ ]:
# 대륙별 주종별 총 서빙량
continents_sum

,beer_servings,spirit_servings,wine_servings
continent,,,
AF,3258,866,862
AS,1630,2677,399
EU,8720,5965,6400
NA,3345,3812,564
OC,1435,935,570
SA,2101,1377,749


In [ ]:
# 대륙별 주종별 총 서빙량의 평균
continents_sum_mean = drinks[cor_cols].groupby(drinks["continent"]).sum().mean()

In [ ]:
continents_sum_mean

beer_servings      3414.833333
spirit_servings    2605.333333
wine_servings      1590.666667
dtype: float64

### 주종별 가장 높은 대륙

In [ ]:
beverage_cols = ['beer_servings', 'spirit_servings', 'wine_servings']
max_continent_by_bev = {}

for col in beverage_cols:
    # 대륙별 총 서빙량 합계(Series)
    continent_sum = drinks.groupby('continent')[col].sum()
    # 합계가 가장 큰 인덱스(=대륙 코드)를 구함
    max_cont = continent_sum.idxmax()
    max_continent_by_bev[col] = max_cont

for col, cont in max_continent_by_bev.items():
     print(f"- {col} 총 서빙량이 가장 많은 대륙: {cont}")


- beer_servings 총 서빙량이 가장 많은 대륙: EU
- spirit_servings 총 서빙량이 가장 많은 대륙: EU
- wine_servings 총 서빙량이 가장 많은 대륙: EU


In [ ]:
alc_mean = drinks["total_litres_of_pure_alcohol"].mean()

In [ ]:
alc_mean

4.717098445595855

In [ ]:
con_mean = drinks.groupby("continent")["total_litres_of_pure_alcohol"].mean()

In [ ]:
con_mean

continent
AF    3.007547
AS    2.170455
EU    8.617778
NA    5.995652
OC    3.381250
SA    6.308333
Name: total_litres_of_pure_alcohol, dtype: float64

In [ ]:
# 전체 평균 이상인 대륙 (하위 3대륙 - 유럽 > 북미 > 남미)
con_mean[con_mean >= alc_mean]

continent
EU    8.617778
NA    5.995652
SA    6.308333
Name: total_litres_of_pure_alcohol, dtype: float64

In [ ]:
# 전체 평균 이하인 대륙 (하위 3대륙 - 오세아니아 > 아프리카 > 아시아)
con_mean[con_mean <= alc_mean]

continent
AF    3.007547
AS    2.170455
OC    3.381250
Name: total_litres_of_pure_alcohol, dtype: float64

# 주종별 유럽의 상위 3국가, 하위 3국가

### beer_servings 가 가장 높은 대륙

In [ ]:
max_beer = drinks.groupby("continent")["beer_servings"].sum().idxmax() # 최댓값의 맥스를 뽑아줌

In [ ]:
max_beer

'EU'

In [ ]:
# 유럽(EU) 내에서 1인당 맥주 서빙 횟수가 가장 높은 상위 3개 국가
Top3_EU_beer = drinks[drinks["continent"] == max_beer].sort_values(by="beer_servings", ascending=False)[["country", "beer_servings"]].head(3).reset_index(drop=True)

In [ ]:
Top3_EU_beer

,country,beer_servings
0,Czech Republic,361
1,Germany,346
2,Poland,343


In [ ]:
Bot3_EU_beer = drinks[drinks["continent"] == max_beer].sort_values(by="beer_servings", ascending=False)[["country", "beer_servings"]].tail(3).reset_index(drop=True)

In [ ]:
# 하위 3국가는 데이터가 0인 것들이 있어서 그냥 발표 참고용으로 이런 국가가 있다~ 라고 말하시거나 그냥 빼도 괜찮을 것 같아요?
Bot3_EU_beer

,country,beer_servings
0,Armenia,21
1,San Marino,0
2,Monaco,0


### wine_servings 가 가장 높은 대륙

In [ ]:
max_wine = drinks.groupby("continent")["wine_servings"].sum().idxmax()

In [ ]:
max_wine

'EU'

In [ ]:
# 유럽(EU) 내에서 1인당 와인 서빙 횟수가 가장 높은 상위 3개 국가
Top3_EU_wine = drinks[drinks["continent"] == max_beer].sort_values(by="wine_servings", ascending=False)[["country", "wine_servings"]].head(3).reset_index(drop=True)

In [ ]:
Top3_EU_wine

,country,wine_servings
0,France,370
1,Portugal,339
2,Andorra,312


In [ ]:
# 하위 3
Bot3_EU_wine = drinks[drinks["continent"] == max_wine].sort_values(by="wine_servings", ascending=False)[["country", "wine_servings"]].tail(3).reset_index(drop=True)

In [ ]:
Bot3_EU_wine

,country,wine_servings
0,Azerbaijan,5
1,San Marino,0
2,Monaco,0


### spirit_servings 가 가장 높은 대륙

In [ ]:
max_spirit = drinks.groupby("continent")["spirit_servings"].sum().idxmax()

In [ ]:
max_spirit

'EU'

In [ ]:
# 유럽(EU) 내에서 1인당 증류주 서빙 횟수가 가장 높은 상위 3개 국가
Top3_EU_spirit = drinks[drinks["continent"] == max_beer].sort_values(by="spirit_servings", ascending=False)[["country", "spirit_servings"]].head(3).reset_index(drop=True)

In [ ]:
Top3_EU_spirit

,country,spirit_servings
0,Belarus,373
1,Slovakia,293
2,Bulgaria,252


In [ ]:
# 하위 3
Bot3_EU_spirit = drinks[drinks["continent"] == max_spirit].sort_values(by="spirit_servings", ascending=False)[["country", "spirit_servings"]].tail(3).reset_index(drop=True)

In [ ]:
Bot3_EU_spirit

,country,spirit_servings
0,Macedonia,27
1,San Marino,0
2,Monaco,0
